packages/ckeditor5-image/src/imageupload/utils.js

```diff
@@ -82,3 +85,81 @@ function getImageMimeType( blob, src ) {
 		return 'image/jpeg';
 	}
 }
+
+/**
+ * Creates integrations object that will be passed to the
+ * {@link module:image/imageupload/ui/imageuploadpanelview~ImageUploadPanelView}.
+ *
+ * @param {module:core/editor/editor~Editor} editor Editor instance.
+ *
+ * @returns {Object.<String, module:ui/view~View>} Integrations object.
+ */
+
+export function prepareIntegrations( editor ) {
+	const panelItems = editor.config.get( 'image.upload.panel.items' );
+	const imageUploadUIPlugin = editor.plugins.get( 'ImageUploadUI' );
+
+	const PREDEFINED_INTEGRATIONS = {
+		'insertImageViaUrl': createLabeledInputView( editor.locale )
+	};
+
+	if ( !panelItems ) {
+		return PREDEFINED_INTEGRATIONS;
+	}
+
+	// Prepares ckfinder component for the `openCKFinder` integration token.
+	if ( panelItems.find( item => item === 'openCKFinder' ) && editor.ui.componentFactory.has( 'ckfinder' ) ) {
+		const ckFinderButton = editor.ui.componentFactory.create( 'ckfinder' );
+		ckFinderButton.set( {
+			withText: true,
+			class: 'ck-image-upload__ck-finder-button'
+		} );
+
+		// We want to close the dropdown panel view when user clicks the ckFinderButton.
+		ckFinderButton.delegate( 'execute' ).to( imageUploadUIPlugin, 'cancel' );
+
+		PREDEFINED_INTEGRATIONS.openCKFinder = ckFinderButton;
+	}
+
+	// Creates integrations object of valid views to pass it to the ImageUploadPanelView.
+	const integrations = panelItems.reduce( ( object, key ) => {
+		if ( PREDEFINED_INTEGRATIONS[ key ] ) {
+			object[ key ] = PREDEFINED_INTEGRATIONS[ key ];
+		} else if ( editor.ui.componentFactory.has( key ) ) {
+			object[ key ] = editor.ui.componentFactory.create( key );
+		} else {
+			// Console.warn should be enough because missing integration doesn't break the editor.
+			console.warn(
+				'It looks like integration "' + key + '" doesn\'t exist. ' +
+				'What may have happened: ' +
+				'\n- you passed the invalid integration name in the `image.upload.panel.items`,' +
+				'\n- integration component wasn\'t properly registered by the plugin,' +
+				'\n- the plugin that registers the component wasn\'t installed in the editor.'
+			);
+		}
+
+		return object;
+	}, {} );
+
+	return integrations;
```

Comment:
```suggestion
	// Creates integrations object of valid views to pass it to the ImageUploadPanelView.
	return panelItems.reduce( ( object, key ) => {
		if ( PREDEFINED_INTEGRATIONS[ key ] ) {
			object[ key ] = PREDEFINED_INTEGRATIONS[ key ];
		} else if ( editor.ui.componentFactory.has( key ) ) {
			object[ key ] = editor.ui.componentFactory.create( key );
		} else {
			/**
			 * The specified name of the view cannot be created by the
			 * {@link module:ui/componentfactory~ComponentFactory component factory}.
			 *
			 * Check whether:
			 *  * you passed the proper integration name as the
			 *    `{@link module:image/imageupload~ImageUploadPanelConfig#items image.upload.panel.items}`,
			 *  * the plugin that registers the component was loaded to the editor,
			 *  * integration component was properly registered by the plugin.
			 *
			 * @error image-upload-integrations-invalid-view
			 */
			console.warn( attachLinkToDocumentation(
				'image-upload-integrations-invalid-view: Trying to use a view that does not exist.'
			), { key } );
		}

		return object;
	}, {} );
```

1. You don't need the `integrations` variable,
2. We're describing warnings like errors.

After applying the suggestion, don't forget to import the `attachLinkToDocumentation()` util.

```js
import { attachLinkToDocumentation } from '@ckeditor/ckeditor5-utils/src/ckeditorerror';
```